src/System.Slices/src/System/MemoryUtils.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    internal static class MemoryUtils
+    {
+        /// <summary>
+        /// platform independent fast memory comparison
+        /// for x64 it is as fast as memcmp of msvcrt.dll, for x86 it is up to two times faster!!
+        /// </summary>
+        internal static bool MemCmp<[Primitive]T>(Span<T> first, Span<T> second)
+            where T : struct
+        {
+            unsafe
+            {
+                // prevent GC from moving memory
+                var firstPinnedHandle = GCHandle.Alloc(first.Object, GCHandleType.Pinned);
```

Comment:
is pinning worth for short spans? I'm not sure how expensive is doing it (could be close to nop too although worth checking)